### PR TITLE
Improve voting UI

### LINKED ISSUE #837: Improve voting UI
1. Dont make vote window disappear after voting.
2. Show the amount of for/against votes in the voting window.
3. Allow admins to force votes they've already voted in. It's annoying when you cant cancel a vote because you accidentally pressed P instead of Ctrl+P
4. Remove "asda voted for" "adsa voted against" chat spam, but keep some way of logging who voted for/against.
5. Make it possible to change your vote? May not be a good idea

### PR DESCRIPTION
**Pull request by laws65**
## Status
**READY**: this PR is (to the best of your knowledge) ready to be incorporated into the game.

## Description
Resolves #837
- Stops the vote window from disappearing when you vote
- Allows admins to force votes they've already voted in
- Shows the amount of votes for and against in the vote window
- Removes "player voted for/against" chat spam (still shown for admins, devs, guards, and the server logs)

## Steps to Test
As an admin:
- Start a vote
- Vote normally for/against
- See that there's a chat message about your vote
- Force vote for/against

As a normal player:
- Start a vote
- Vote for something
- See that vote window shows your vote
- See that there's no chat message about your vote
- Try to make a second vote
- Click on the vote menu to make it disappear

### CODE REVIEW

**Comment by laws65** [comment]
Here's what it looks like:
image

File: Rules/CommonScripts/VoteCommon.as [hunk 1]
```diff
@@ -221,13 +221,26 @@ void Vote(VoteObject@ vote, CPlayer@ p, bool favour)
...
 		{
-
-
-			client_AddToChat(getTranslatedString("--- {USER} Voted {DECISION} ---").replace("{USER}", p.getUsername()).replace("{DECISION}", getTranslatedString(favour ? "In Favour" : "Against")), vote_message_colour());
+			should_show_votes = getNet().isServer();
+		}
+		else
+		{
+			should_show_votes = (player.isDev() || player.isGuard() || getNet().isServer()
+			|| getSecurity().checkAccess_Feature(player, "admin_color") || player.isRCON());
```
Comment (CatWiz) [inline Rules/CommonScripts/VoteCommon.as:234]: Wouldn't it make more sense to check for player.isMod() here instead of admin_color?

Comment (laws65) [reply Rules/CommonScripts/VoteCommon.as:234]: Yep, I'll fix that

File: Rules/CommonScripts/VoteCommon.as [hunk 2]
```diff
@@ -221,13 +221,26 @@ void Vote(VoteObject@ vote, CPlayer@ p, bool favour)
...
-		if (CanPlayerVote(vote, getLocalPlayer()) || getNet().isServer()) //include all in server logs
+		
+		bool should_show_votes;
+		CPlayer@ player = getLocalPlayer();
+		if (player is null)
 		{
-
-
-			client_AddToChat(getTranslatedString("--- {USER} Voted {DECISION} ---").replace("{USER}", p.getUsername()).replace("{DECISION}", getTranslatedString(favour ? "In Favour" : "Against")), vote_message_colour());
+			should_show_votes = getNet().isServer();
```
Comment (Vam-Jam) [inline Rules/CommonScripts/VoteCommon.as:229]: Use isServer() instead of getNet().isServer()

File: Rules/CommonScripts/VoteCommon.as [hunk 3]
```diff
@@ -221,13 +221,26 @@ void Vote(VoteObject@ vote, CPlayer@ p, bool favour)
...
+		}
+		else
+		{
+			should_show_votes = (player.isDev() || player.isGuard() || getNet().isServer()
```
Comment (Vam-Jam) [inline Rules/CommonScripts/VoteCommon.as:233]: Same here

**Review by AsuMagic — APPROVED** [review]

**Comment by laws65** [comment]
Merge when

**Pull request merged**